variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/VariantClassifier.java

```diff
@@ -33,20 +35,28 @@
      * @return Type of the variant
      */
     public static VariantType getVariantClassification(String reference, String alternate, int subSNPClass) {
-        String alphaRegEx = "[A-Z]+";
+        Pattern alphaRegExPattern = Pattern.compile("[A-Z]+");
 
         reference = reference.trim();
         alternate = alternate.trim();
 
-        if (!reference.equals("NOVARIATION")) {
+        boolean isRefAlpha = alphaRegExPattern.matcher(reference).matches();
+        boolean isAltAlpha = alphaRegExPattern.matcher(alternate).matches();
+
+        if (reference.equals("NOVARIATION")) {
+            if (subSNPClass == 6) {
+                return VariantType.NO_SEQUENCE_ALTERATION;
+            }
+        }
+        else {
             if (subSNPClass == 4) {
                 return VariantType.TANDEM_REPEAT;
             }
             if (subSNPClass == 5) {
                 return VariantType.SEQUENCE_ALTERATION;
             }
             if (!alternate.equals(reference)) {
-                if (reference.matches(alphaRegEx) && alternate.matches(alphaRegEx)) {
+                if (isRefAlpha && isAltAlpha) {
```

Comment:
This part of the code is difficult to read, in part because of how the 'else' have been written. I think some refactor would be good here, maybe extracting some private method (even if this method is not so big).